src/main/resources/webgoat/templates/main_new.html

```diff
@@ -151,6 +152,8 @@
                 </button>
             </a>
 
+            <input class="form-control" type="text" id="search" name="search" th:placeholder="#{searchmenu}" onchange="search(this.value);" style="width:60%" />
```

Comment:
I tested it and with onchange is triggered when blurred. What do you think to use onkeypress instead will work like as an instant search; if don't like it, we can add something like `if (e.key === 'Enter') {` if we want to press enter for triggering the search.

Reply:
I changed on change to on input with a minimum of 3 chars and it writes the lesson description back to the value field.

Reply:
Do you like this better or is the last part exaggerated

Reply:
I think it's a good idea, but I've been testing it and sometimes it makes the change too fast when I am typing. For example:
(searching for "injection"):
sql **inj**ection (mitigation)**ec**
(searching for "spoof"):
**spo**ofing an authentication cookie**o**

Reply:
Thanks or testing it and giving feedback. Do you think it is intuitive and good enough as a first introduction of this search functionality or do we need something more?

Reply:
You're welcome. I think it's a great idea and great functionality. The only thing that happened to me is the completion feature is error prone because is too fast. Maybe we can improve applying some ideas I have in mind:
- remove the autocomplete and let the field as it is
- or keep the autocomplete but maybe using autocomplete gray text, preserving the current text and only replace it pressing , for example, enter
- or use the autocomplete feature like a drop down showing the all occurrences instead of only the first one (the current array[0]).

What do you think?